client/proto/state_snapshot/v1/state_snapshot.proto

```diff
@@ -0,0 +1,72 @@
+// Copyright 2021 VMware, all rights reserved
+//
+// Concord Client's State Snapshot Service
+
+syntax = "proto3";
+package vmware.concord.client.snapshot.v1;
+
+import "google/protobuf/empty.proto";
+import "google/protobuf/wrappers.proto";
+
+option java_package = "com.vmware.concord.client.snapshot.v1";
+
+// The StateSnapshotService can be used to initialize a client application from a
+// state snapshot visible to Concord Client. State snapshots aggregate all writes
+// from the start of the blockchain up to a specific point by keeping only the last
+// write to a key-value.
+//
+// Concord Client ensures that there is always at least one reasonably
+// recent snapshot that can be served. The intention is that this snapshot is
+// used to initialize a client application after the network has been pruned and
+// some set of historical events are no longer available via the EventService.
+service StateSnapshotService {
+  // Get the ID of a recent and available snapshot.
+  // Errors:
+  // UNAVAILABLE: if Concord Client is not ready yet to process requests.
+  // UNKNOWN: exact cause is unknown.
+  rpc GetRecentStateSnapshotId(google.protobuf.Empty) returns (StateSnapshotId);
+
+  // Stream a specific state snapshot in a resumable fashion as a stream of key-values.
+  // Key-values are streamed with lexicographic order on keys.
```

Comment:
@miklos-da: let's double-check that 

1. packages are always ordered before contracts in the protobuf encoding we use 
2. we have the ability to introduce further keys ordered before packages, in case we need to serve some data before packages in future extensions (e.g., package metadata)

Reply:
thx. LGTM. You can resolve this comment.